### PR TITLE
Add Pekko support

### PR DESCRIPTION
**Pull request by benjamingeer**

### CODE REVIEW

**Comment by kciesielski** [comment]
Thanks @benjamingeer, looks really promising.

File: server/pekko-http-server/src/main/scala/sttp/tapir/server/pekkohttp/PekkoBodyListener.scala [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      case Right(r) =>
+        Future.successful(Right(r.transformDataBytes(Flow[ByteString].watchTermination() { case (_, f) =>
+          f.onComplete {
+            case Failure(ex) => cb(Failure(ex))
+            case Success(_)  => cb(Success(()))
+          }
+        })))
+    }
+  }
+}
```
Comment (shinharad) [inline server/pekko-http-server/src/main/scala/sttp/tapir/server/pekkohttp/PekkoBodyListener.scala:24]: @benjamingeer I would like to propose a suggestion. Could you please apply the changes from `AkkaBodyListener` into `PekkoBodyListener`? https://github.com/softwaremill/tapir/pull/2880

**Comment by He-Pin** [comment]
@benjamingeer It's released, any update on this?

**Comment by kciesielski** [comment]
@He-Pin This PR still needs `sttp` to support Pekko. The support there is implemented and ready to merge for the master branch: https://github.com/softwaremill/sttp/pull/1781, then we need to port it to the 3.x branch. I suspect it will all be ready in a few days at most.

File: server/pekko-http-server/src/main/scala/sttp/tapir/server/pekkohttp/PekkoBodyListener.scala [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+import org.apache.pekko.stream.scaladsl.Flow
+import org.apache.pekko.util.ByteString
+import sttp.tapir.server.interpreter.BodyListener
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.util.{Failure, Success, Try}
+
+class PekkoBodyListener(implicit ec: ExecutionContext) extends BodyListener[Future, PekkoResponseBody] {
```
[the remark this answers is not in the document]
Comment (sergionoviello) [reply server/pekko-http-server/src/main/scala/sttp/tapir/server/pekkohttp/PekkoBodyListener.scala:11]: hi @kciesielski I've added the same fix as https://github.com/softwaremill/tapir/pull/2880
Let me know if this is ok.

**Comment by sergionoviello** [comment]
> @He-Pin This PR still needs `sttp` to support Pekko. The support there is implemented and ready to merge for the master branch: [softwaremill/sttp#1781](https://github.com/softwaremill/sttp/pull/1781), then we need to port it to the 3.x branch. I suspect it will all be ready in a few days at most.

@kciesielski what's the next step. Are you ok to take over and get this merged when sttp support to Pekko is ready?

**Comment by kciesielski** [comment]
@sergionoviello Yes, I'm taking over, working on sttp3 right now and then will get back here to finish this PR :)

**Comment by kciesielski** [comment]
And thanks a lot for adding the listner fix @sergionoviello!

File: examples/src/main/scala/sttp/tapir/examples/HelloWorldPekkkoServer.scala
Comment (pjfanning) [inline examples/src/main/scala/sttp/tapir/examples/HelloWorldPekkkoServer.scala:1]: The file name has 3 k's in `Pekkko`.

Comment (kciesielski) [reply examples/src/main/scala/sttp/tapir/examples/HelloWorldPekkkoServer.scala:1]: Thanks, fixed :)

**Pull request merged**